hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/operations/PartitionIteratingOperation.java

```diff
@@ -32,73 +32,73 @@
 import com.hazelcast.util.ResponseQueueFactory;
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collections;
-import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.BlockingQueue;
 
+import static com.hazelcast.util.CollectionUtil.toIntArray;
+
 public final class PartitionIteratingOperation extends AbstractOperation implements IdentifiedDataSerializable {
 
-    private List<Integer> partitions;
     private OperationFactory operationFactory;
-    private Map<Integer, Object> results;
+    private int[] partitions;
+    private Object[] results;
 
-    public PartitionIteratingOperation(List<Integer> partitions, OperationFactory operationFactory) {
-        this.partitions = partitions != null ? partitions : Collections.<Integer>emptyList();
-        this.operationFactory = operationFactory;
+    public PartitionIteratingOperation() {
     }
 
-    public PartitionIteratingOperation() {
+    public PartitionIteratingOperation(OperationFactory operationFactory, List<Integer> partitions) {
```

Comment:
I followed up the call hierarchy and found no easy way of directly create the partitions as `int[]`, so I left the constructor as it is for now.

The `PartitionIteratingOperation` is created in just a single place, generated by looping over a `Map<Address, List<Integer>> memberPartitions`. I had a brief look how that map is created and found no elegant way to directly create a `Map<Address, int[]>`, since the array lengths are not predictable.

So I could easily pull the conversion up a single level. But then we have to suppress this FindBugs issue without having a lot gained:
```
new com.hazelcast.spi.impl.operationservice.impl.operations.PartitionIteratingOperation(OperationFactory, int[]) may expose internal representation by storing an externally mutable object into PartitionIteratingOperation.partitions [com.hazelcast.spi.impl.operationservice.impl.operations.PartitionIteratingOperation] At PartitionIteratingOperation.java:[line 49]
```
So for now I would leave this as it is, until we find a way to cleanup the whole call hierarchy.